Source/santad/EventProviders/SNTEndpointSecurityProcessFileAccessAuthorizer.mm

```diff
@@ -121,28 +121,21 @@ - (void)handleMessage:(Message &&)esMsg
 
   switch (esMsg->event_type) {
     case ES_EVENT_TYPE_NOTIFY_EXEC: {
-      // If the exec was allowed, remove the pid/pidver of the process that
-      // was just replaced. If the exec was denied, cleanup the entry that
-      // was added optimistically in the probe. The EXIT event will take care
-      // of the original entry later on.
-      if (esMsg->action.notify.result.auth == ES_AUTH_RESULT_ALLOW) {
-        _procRuleCache->remove(PidPidversion(esMsg->process->audit_token));
-      } else {
-        _procRuleCache->remove(PidPidversion(esMsg->event.exec.target->audit_token));
-      }
-
+      // On EXEC, the previous process was replaced with the new one. The old
+      // process must be cleaned up since there will be no EXIT event for it.
+      // There will be a corresponding EXIT for the newly executed process
+      // regardless of whether or not it was allowed or denied.
```

Comment:
This comment is a little confusing to me. IIUC the process wasn't replaced, just the pid version was incremented.